components/bin/build

```diff
@@ -0,0 +1,318 @@
+ */
+function processParts(parts) {
+    const objects = [];
+    for (let i = 1; i < parts.length; i += 2) {
+        const words = parts[i].split(/\s+/);
+        const type = words[words.length - 2];
+        const name = words[words.length - 1];
+
+        if (words[1] === 'default') {
+
+            objects.push('default');
+
+        } else {
+
+            switch (type) {
+              case 'type':
+              case 'interface':
+                break;
+                
+              case 'let':
+              case 'const':
+              case 'var':
+              case 'function':
+              case 'class':
+              case 'namespace':
+                objects.push(name);
+                break;
+                
+              case 'default':
+                objects.push('default');
+                break;
+                
+              default:
+                console.log("   Can't process '" + parts[i] + "'");
+                break;
+            }
+        }
+    }
+    return objects;
+}
+
+/**
+ * Create the lines of the lib file using the object names from the file.
+ *
+ * @param {string} name       The path of the file relative to the root .ts directory
+ * @param {string[]} objects  Array of names of the object exported by the file
+ * @return {string[]}         Array of lines for the contents of the library file
+ */
+function processLines(file, objects) {
+    if (objects.length === 0) return [];
+    const dir = path.dirname(file).replace(/^\.$/, '');
+    const dots = dir.replace(/[^\/]+/g, '..') || '.';
+    const relative = path.join(dots, '..', MATHJAX3, dir, path.basename(file)).replace(/\.ts$/, '.js');
+    const name = path.parse(file).name;
+    const lines = [
+        '"use strict";',
+        `Object.defineProperty(exports, '__esModule', {value: true});`
+    ];
+    let source = (dir.replace(/\//g, '.') + '.' + name).replace(/^\./, '')
+        .replace(/\.[^.]*/g, (x) => (x.substr(1).match(/[^a-zA-Z_]/) ? '[\'' + x.substr(1) + '\']' : x));
+    if (exists(path.resolve(MATHJAX3, file.replace(/\.ts$/, '')))) source += '_ts';
+    for (const id of objects) {
+        lines.push(`exports.${id} = MathJax._.${source}.${id};`);
+    }
+    return lines;
+}
+
+/**
+ * @param {string} file     Path to a file
+ * @return {boolean}        True if the file exists, false if not
+ */
+function exists(file) {
+    if (!fs.existsSync(file)) return false;
+    //
+    // For case-insensitive filing systems (like Mac OS),
+    //   check that the file names match exactly
+    //
+    const [dir, name] = [path.dirname(file), path.basename(file)];
+    return fs.readdirSync(dir).indexOf(name) >= 0;
+}
+
+/**
+ * Recursively make any needed directories
+ *
+ * @param {string} dir   The directory relative to the library directory
+ */
+function makeDir(dir) {
+    const fulldir = path.resolve(LIB, dir);
+    if (fs.existsSync(fulldir)) return;
+    makeDir(path.dirname(fulldir));
+    fs.mkdirSync(fulldir);
```

Comment:
Could we just use `shelljs` here?

Reply:
I think this is simple enough not to require an additional dependency.